modules/nf-core/bcftools/pluginsetgt/main.nf

```diff
@@ -0,0 +1,76 @@
+process BCFTOOLS_PLUGINSETGT {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/bcftools:1.21--h8b25389_0':
+        'biocontainers/bcftools:1.21--h8b25389_0' }"
+
+    input:
+    tuple val(meta), path(vcf), path(index)
+    path(regions)
+    path(targets)
+
+    output:
+    tuple val(meta), path("*.{vcf,vcf.gz,bcf,bcf.gz}"), emit: vcf
+    tuple val(meta), path("*.tbi")                    , emit: tbi, optional: true
+    tuple val(meta), path("*.csi")                    , emit: csi, optional: true
+    path "versions.yml"                               , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def args2 = task.ext.args2 ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def regions_file = regions ? "--regions-file ${regions}" : ""
+    def targets_file = targets ? "--targets-file ${targets}" : ""
+    def extension = args.contains("--output-type b") || args.contains("-Ob") ? "bcf.gz" :
+                    args.contains("--output-type u") || args.contains("-Ou") ? "bcf" :
+                    args.contains("--output-type z") || args.contains("-Oz") ? "vcf.gz" :
+                    args.contains("--output-type v") || args.contains("-Ov") ? "vcf" :
+                    "vcf"
+    """
+    bcftools plugin setGT \\
+        --output ${prefix}.${extension} \\
+        ${regions_file} \\
+        ${targets_file} \\
+        $args \\
+        --threads $task.cpus \\
+        ${vcf} \\
+        -- \\
+        $args2
```

Comment:
What's this doing?

Reply:
BCFtools plugins are run as bcftools [plugin NAME|+NAME] [OPTIONS] FILE  --  [PLUGIN OPTIONS]. Most plugins accept two types of parameters: general options shared by all plugins followed by a separator, and a list of plugin-specific options. I've set `args` to be the general options and `args2` to be the plugin-specific options.

Reply:
Allright thanks! That seems reasonable then :)

Reply:
Does it also work if you don't give any args in `args2`?

Reply:
It does not, at minimum the `-n, --new-gt TYPE` and ` -t, --target-gt TYPE ` plugin options need to be set for this plugin to work

Reply:
non-file mandatory command arguments added here https://github.com/nf-core/modules/pull/6718/commits/a5e6c0cdd17b2581b7a1b969926f2d8c0d08f3a9